src/Compilers/CSharp/Portable/Symbols/Tuples/TupleErrorFieldSymbol.cs

```diff
@@ -124,7 +124,10 @@ public bool Equals(TupleErrorFieldSymbol other)
                 return true;
             }
 
-            return (object)other != null && _tupleFieldId == other._tupleFieldId && ContainingType == other.ContainingType;
+            return (object)other != null &&
+                _tupleElementIndex == other._tupleElementIndex &&
+                ContainingType == other.ContainingType &&
+                Name == other.Name;
```

Comment:
In `TupleTypeSymbol.CreateMembers`, we can create two error field types, one called "Item1" and the other called "a". Previously, we created them with different tuple element indices (and therefore they were not equal). Now, we create them with the correct index (0), but they should still test as not equal.